### PR TITLE
cliam: init at 1.0.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Description of changes
Cloud agnostic IAM permissions enumerator

https://github.com/securisec/cliam

Related to https://github.com/NixOS/nixpkgs/issues/81418

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/security/cliam/default.nix [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    owner = "securisec";
+    repo = pname;
+    rev = version;
+    hash = "sha256-bq7u6pknokyY4WwO1qMYPuY86UZlDgeYEa1AJpk8d+4=";
+  };
+
+  vendorSha256 = "sha256-aGBA97EvIUv9myqcrtltiVxh1/0VtrQy2j9GU6r197g=";
+
+  ldflags = [
+    "-s -w"
```
Comment (06kellyjac) [inline pkgs/tools/security/cliam/default.nix:20]: does this work?..

```suggestion
    "-s"
    "-w"
```

File: pkgs/tools/security/cliam/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    "-X github.com/securisec/cliam/cli/version.Version=${version}"
+  ];
+
+  postInstall = ''
+    mv $out/bin/cli $out/bin/$pname
+  '';
```
Comment (06kellyjac) [inline pkgs/tools/security/cliam/default.nix:26]: this also works:

```suggestion
  postBuild = ''
    # should be called cliam
    mv $GOPATH/bin/{cli,cliam}
  '';
```

File: pkgs/tools/security/cliam/default.nix [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
```
Comment (06kellyjac) [inline pkgs/tools/security/cliam/default.nix:27]: cliam looks like it provides completions according to the readme. might be worth adding something like this:

```suggestion
  postInstall = ''
    installShellCompletion --cmd cliam \
      --bash <($out/bin/cliam completion bash) \
      --fish <($out/bin/cliam completion fish) \
      --zsh <($out/bin/cliam completion zsh)
  '';
```

File: pkgs/tools/security/cliam/default.nix [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  meta = with lib; {
+    description = "Cloud agnostic IAM permissions enumerator";
+    homepage = "https://github.com/securisec/cliam";
+    license = licenses.gpl3Only;
```
Comment (06kellyjac) [inline pkgs/tools/security/cliam/default.nix:31]: do we need to check if this is Only or Plus?

File: pkgs/tools/security/cliam/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/tools/security/cliam/default.nix:20]: Updated.

File: pkgs/tools/security/cliam/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/tools/security/cliam/default.nix:26]: Thanks, updated.

File: pkgs/tools/security/cliam/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/tools/security/cliam/default.nix:27]: Added, thanks.

File: pkgs/tools/security/cliam/default.nix (hunk 4 above)
Comment (fabaff) [reply pkgs/tools/security/cliam/default.nix:31]: There is no mentioning of "any later...". Thus, we assume that it's "Only".

**Review by 06kellyjac — APPROVED** [review]

**Comment by fabaff** [comment]
@06kellyjac, thanks for your feedback.

**Pull request merged**